shared/src/main/scala/com/comcast/ip4s/IDN.scala

```diff
@@ -38,7 +35,7 @@ import scala.util.hashing.MurmurHash3
   * Note: equality and comparison of IDNs is case-sensitive. Consider comparing normalized toString values
   * for a more lenient notion of equality.
   */
-final class IDN private (val labels: NonEmptyList[IDN.Label], val hostname: Hostname, override val toString: String)
+final class IDN private (val labels: List[IDN.Label], val hostname: Hostname, override val toString: String)
```

Comment:
Before the cats dependency, this was basically a `(IDN.Label, List[IDN.Label])`. Not sure if that's overkill or not though, as often folks will ignore the separate components and instead just want to work with the labels as a list.